api/v1alpha3/virtualmachine_types.go

```diff
@@ -420,6 +420,27 @@ type VirtualMachineSpec struct {
 
 	// +optional
 
+	// EncryptionClassName describes the name of the EncryptionClass resource
+	// used to encrypt this VM.
+	//
+	// Please note, this field is not required to encrypt the VM. If the
+	// underlying platform implements the vSphere Native Key Provider (NKP) then
+	// the VM may still be fully or partially encrypted depending on the
+	// specified storage and VM classes.
+	//
+	// If an NKP is present and an encryption storage class is selected, the
+	// VM's home files and non-PVC disks will be encrypted.
```

Comment:
Maybe its worth noting somewhere that in case there is a default key provider in vCenter AND EncryptionClassName (ECN) is specified, then ECN takes precedence.